src/ACBr.Net.NFSe/Providers/SaoPaulo/ProviderSaoPaulo.cs

```diff
@@ -62,25 +62,28 @@ public ProviderSaoPaulo(ConfigNFSe config, ACBrMunicipioNFSe municipio) : base(c
         public override NotaFiscal LoadXml(XDocument xml)
         {
             Guard.Against<XmlException>(xml == null, "Xml invalido.");
+            
+            XElement rootDoc = xml.Root;
+            Guard.Against<XmlException>(rootDoc == null, "Xml de RPS ou NFSe invalido.");
 
             var ret = new NotaFiscal();
             ret.Assinatura = xml.Root?.ElementAnyNs("Assinatura")?.GetValue<string>() ?? string.Empty;
```

Comment:
Utilize a variável rootDoc aqui também.
```suggestion
            ret.Assinatura = rootDoc.ElementAnyNs("Assinatura")?.GetValue<string>() ?? string.Empty;
```